packages/flutter_tools/test/commands.shard/hermetic/drive_test.dart

```diff
@@ -0,0 +1,68 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// @dart = 2.8
+
+import 'package:file/memory.dart';
+import 'package:flutter_tools/src/base/file_system.dart';
+import 'package:flutter_tools/src/base/logger.dart';
+import 'package:flutter_tools/src/base/platform.dart';
+import 'package:flutter_tools/src/commands/drive.dart';
+import 'package:flutter_tools/src/device.dart';
+import 'package:test/fake.dart';
+
+import '../../src/common.dart';
+
+void main() {
+  FileSystem fileSystem;
+  BufferLogger logger;
+  Platform platform;
+
+  setUp(() {
+    fileSystem = MemoryFileSystem.test();
+    logger = BufferLogger.test();
+    platform = FakePlatform(operatingSystem: 'linux');
+  });
+
+  testWithoutContext('drive --screenshot writes to expected output', () async {
+    final Device screenshotDevice = ScreenshotDevice();
+
+    await takeScreenshot(
+      screenshotDevice,
+      'drive_screenshots',
+      fileSystem,
+      logger,
+      FileSystemUtils(
+        fileSystem: fileSystem,
+        platform: platform,
+      ),
+    );
+
+    expect(logger.statusText, contains('Screenshot written to drive_screenshots/drive_01.png'));
+  });
+
+  testWithoutContext('drive --screenshot errors but does not fail if screenshot fails', () async {
```

Comment:
I tried for awhile to get `DriveCommand` working with the `CommandRunner` to actually run the test, but I eventually gave up after I couldn't pass in `--no-pub`
https://github.com/flutter/flutter/blob/ce318b7b539e228b806f81b3fa7b33793c2a2685/packages/flutter_tools/lib/src/commands/drive.dart#L142-L145
https://github.com/flutter/flutter/blob/74bd7b6f6db6433783a36f94d9e43173745187f4/packages/flutter_tools/test/src/throwing_pub.dart#L37

Reply:
hmm, I think this is wrong. We should always run pub by default, unless we're asked not to.